### ISSUE TITLE
Crash on adding coffee beans to trade offer

### ISSUE THREAD

**Issue by mrkybe** [CleverRaven/Cataclysm-DDA#31752]
# Describe the bug

Looks like the same thing as #31559 except its happening with `plastic bag of chocolate-covered coffee beans (fresh) (5/720) (3)`.

# Steps To Reproduce from the included save-game

0. Unzip the saved game to your CDDA dir.  I have a custom vehicle parts mod that I included since I'm not sure if the save will load without you having it.  
1. Talk to NPC south of your starting position.
2. Trade.
3. Try to add the `plastic bag of chocolate-covered coffee beans (fresh) (5/720) (3)` to the trade offer.
4. Crash.

# Expected behavior

Not crash.

# Saved Game

Its `Eden Megacity 1`

[crashy.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/3317391/crashy.zip)

# Versions and configuration

Latest experimental.

 - OS: [e.g. iOS 8 or Windows 10 or Ubuntu 18.04]
 - Game Version: [from the main menu, e.g. 0.C-29938-g90f5268437]
 - Graphics version: [Tiles or Terminal]
 - Mods loaded: [e.g.dda, boats, hacktheplanet, StatsThroughSkills]

# Additional context

**Comment by AMurkin** [comment]
Should be fixed as #31559.

**Comment by AMurkin** [comment]
Doesn't crash after #32392.
Although the fix was incomplete and the input box for the number of items does not display the entered value.
![изображение](https://user-images.githubusercontent.com/1931904/61582861-0bc20780-ab41-11e9-862a-ea872cd10fdc.png)

**Comment by Night-Pryanik** [comment]
I don't know what's going on, but on 0.D-8431 I just can't offer these coffee beans to the NPC.
![изображение](https://user-images.githubusercontent.com/11132525/66705219-02f26380-ed35-11e9-86f5-08b7a761b2cb.png)

![изображение](https://user-images.githubusercontent.com/11132525/66705223-0ab20800-ed35-11e9-9df8-0c16549bc56c.png)

**Comment by Night-Pryanik** [comment]
Crash is resolved, long names not fitting screen issue is still valid, but I'm closing this in favor of #33434.
